src/GGTestUtils.py

```diff
@@ -300,6 +349,86 @@ def wait_for_deployment_till_timeout(
+                                            )
+                                            print(
+                                                f"AWS Deployment Policies: {aws_deployment.get('deploymentPolicies', {})}"
+                                            )
+                                            if 'components' in aws_deployment:
+                                                print(
+                                                    f"Components in deployment: {list(aws_deployment['components'].keys())}"
+                                                )
+                                        except Exception as e:
+                                            print(
+                                                f"Could not get AWS deployment details: {e}"
+                                            )
+
+                                        print(
+                                            f"\nChecking all Greengrass logs for errors..."
+                                        )
+                                        try:
+                                            import subprocess
+                                            # Check all greengrass services
+                                            services = [
+                                                "ggdeploymentd", "iotcored",
+                                                "ggconfigd", "tesd",
+                                                "gghealthd", "ggipcd"
+                                            ]
+                                            for svc in services:
+                                                log_output = subprocess.run(
+                                                    [
+                                                        "journalctl", "-u", svc,
+                                                        "--no-pager", "-n",
+                                                        "50", "--since",
+                                                        "5 minutes ago"
+                                                    ],
+                                                    capture_output=True,
+                                                    text=True,
+                                                    timeout=3)
+                                                if log_output.stdout.strip(
+                                                ) and "-- No entries --" not in log_output.stdout:
+                                                    print(
+                                                        f"\n--- {svc} logs ---")
+                                                    print(log_output.
+                                                          stdout[-1500:])
+
+                                            # Check for any errors in all logs
+                                            error_log = subprocess.run(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Starting a process with a partial executable path https://bandit.readthedocs.io/en/latest/plugins/b607_start_process_with_partial_path.html